ftd/src/ftd2021/p1/parser.rs

```diff
@@ -401,6 +408,71 @@ pub fn parse_file_for_global_ids(data: &str) -> Vec<(String, usize)> {
     }
 }
 
+fn remove_inline_comments(line: &str) -> String {
+    let mut output = String::new();
+
+    let mut chars = line.chars().peekable();
+
+    let mut escape = false;
+
+    let mut count = 0;
+
+    while let Some(c) = chars.next() {
+        if c == '\\' {
+            if !escape {
+                escape = true;
+            }
+
+            count += 1;
+
+            match chars.peek() {
+                Some(nc) => {
+                    if nc == &';' {
+                        continue;
+                    }
+                }
+                None => break,
+            }
+        }
+
+        if c == ';' {
+            if escape {
+                if count % 2 == 0 {
+                    break;
+                } else {
+                    escape = false;
+                    count = 0;
+                }
+            } else {
+                break;
+            }
+        }
+
+        output.push(c);
+    }
+
+    return output.trim().to_string();
+}
+
+#[cfg(test)]
+mod test2 {
+    macro_rules! t {
+        ($s:expr, $t: expr,) => {
+            t!($s, $t)
+        };
+        ($s:expr, $t: expr) => {
+            assert_eq!(super::remove_inline_comments($s), $t)
+        };
+    }
+
+    #[test]
+    fn test_esc() {
+        t!(r#"Hello      ;; displays the hello"#, "Hello");
+        t!(r#"Hello      \;; displays the hello"#, "Hello      ;");
```

Comment:
This is wrong. There is no comment in this line as comment requires two consecutive unescaped `;;`.